src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java

```diff
@@ -209,12 +211,23 @@ public final XSDSchema buildSchemaInternal(FeatureTypeInfo[] featureTypeInfos, S
             }
 
             if (!simple) {
-                // add secondary namespaces from catalog
-                for (NamespaceInfo nameSpaceinfo : catalog.getNamespaces()) {
-                    if (!schema.getQNamePrefixToNamespaceMap().containsKey(nameSpaceinfo.getPrefix())) {
-                        schema.getQNamePrefixToNamespaceMap().put(nameSpaceinfo.getPrefix(),
-                                nameSpaceinfo.getURI());
+                // complex features may belong to different workspaces
+                WorkspaceInfo localWorkspace = LocalWorkspace.get();
```

Comment:
Yeah, in this case we are only liking the namespace so security wise it is not a problem.